src/realsense2-all.cmake

```diff
@@ -0,0 +1,119 @@
+        get_target_property( _input_type ${input_target} TYPE )
+        if( ${_input_type} STREQUAL "INTERFACE_LIBRARY" )
+            set( _input_link_libraries INTERFACE_LINK_LIBRARIES )
+        endif()
+        get_target_property( public_dependencies ${input_target} ${_input_link_libraries} )
+        if( NOT public_dependencies STREQUAL "public_dependencies-NOTFOUND" )
+            foreach( dependency IN LISTS public_dependencies )
+                if( TARGET ${dependency} )
+                    get_target_property(alias ${dependency} ALIASED_TARGET)
+                    if( TARGET ${alias} )
+                        set( dependency ${alias} )
+                    endif()
+                    get_target_property( _type ${dependency} TYPE )
+                    if( ${_type} STREQUAL "STATIC_LIBRARY" )
+                        list( APPEND static_libs ${dependency} )
+                    else()
+                        #message( "   x ${dependency} not a STATIC_LIBRARY (${_type})" )
+                    endif()
+
+                    get_property( library_already_added
+                        GLOBAL PROPERTY _${target_name}_static_bundle_${dependency} )
+                    if( NOT library_already_added )
+                        set_property( GLOBAL PROPERTY _${target_name}_static_bundle_${dependency} ON )
+                        _recursively_collect_dependencies( ${dependency} )
+                    else()
+                        #message( "   x ${dependency} already done" )
+                    endif()
+                else()
+                    #message( "   x ${dependency} not a target" )
+                endif()
+            endforeach()
+        endif()
+        set( static_libs ${static_libs} PARENT_SCOPE )
+    endfunction()
+
+    list( APPEND static_libs ${target_name} )
+    _recursively_collect_dependencies( ${target_name} )
+    list( REMOVE_DUPLICATES static_libs )
+    #message( "---> Dependencies for ${bundle_name}: ${static_libs}" )
+
+    if( CMAKE_CXX_COMPILER_ID MATCHES "^(Clang|GNU)$" )
+
+        set( bundle_path
+                "${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/${CMAKE_STATIC_LIBRARY_PREFIX}${bundle_name}${CMAKE_STATIC_LIBRARY_SUFFIX}" )
+
+        file( WRITE ${CMAKE_BINARY_DIR}/${bundle_name}.ar.in
+            "CREATE ${bundle_path}\n" )
+        
+        foreach( tgt IN LISTS static_libs )
+            file( APPEND ${CMAKE_BINARY_DIR}/${bundle_name}.ar.in
+                "ADDLIB $<TARGET_FILE:${tgt}>\n" )
+        endforeach()
+    
+        file( APPEND ${CMAKE_BINARY_DIR}/${bundle_name}.ar.in "SAVE\n" )
+        file( APPEND ${CMAKE_BINARY_DIR}/${bundle_name}.ar.in "END\n" )
+
+        file( GENERATE
+            OUTPUT ${CMAKE_BINARY_DIR}/${bundle_name}.ar
+            INPUT ${CMAKE_BINARY_DIR}/${bundle_name}.ar.in )
+
+        set( ar_tool ${CMAKE_AR} )
+        if( CMAKE_INTERPROCEDURAL_OPTIMIZATION )
+            set( ar_tool ${CMAKE_CXX_COMPILER_AR} )
+        endif()
+
+        add_custom_command(
```

Comment:
Please add comment

Reply:
What's unclear? This is what actually runs the archive command.